test/commands/extended/get_inputs_test.py

```diff
@@ -818,3 +867,107 @@ def mock_address_generator(ag, start, step=1):
     self.assertEqual(input0, self.addy1)
     self.assertEqual(input0.balance, 86)
     self.assertEqual(input0.key_index, 1)
+
+  def test_start_stop(self):
+    """
+    Using ``start`` and ``stop`` at once.
+    Checking if correct number of addresses is returned. Must be stop - start
+    """
+    # mocking get_balances to get number of returned balances
+    # equal to number of addresses
+    # returns {"balances": [11] } for one address, {"balances": [11, 11]} for two etc
+
+    def mock_get_balances_execute(adapter, request):
+      return dict(balances=[11] * len(request["addresses"]))
+
+    # To keep the unit test nice and speedy, we will mock the address
+    # generator.  We already have plenty of unit tests for that
+    # functionality, so we can get away with mocking it here.
+    # noinspection PyUnusedLocal
+
+    def mock_address_generator(ag, start, step=1):
+      # returning up to 3 addresses, depending on stop value
+      for addy in [self.addy0, self.addy1, self.addy2][start::step]:
+        yield addy
+
+    with mock.patch(
+        'iota.crypto.addresses.AddressGenerator.create_iterator',
+        mock_address_generator,
+    ):
+      with mock.patch(
+        'iota.commands.core.GetBalancesCommand._execute',
+        mock_get_balances_execute,
+      ):
```

Comment:
💡 If you'd prefer to avoid mocking private methods, here's the way the other unit tests for `GetInputsCommand` simulate API responses:

```python
def test_start(self):
  ...
    self.adapter.seed_response('getBalances', {
      'balances': [86],
    })
    ...
```

`GetInputsCommandTestCase.setup()` injects a `MockAdapter` into the command object, which allows you to specify what response it should simulate for different commands.

Reply:
I'd prefer to avoid mocking private methods and I understand the method of doing it with `seed_response`. In this case the goal was to make getBalances respond dynamically depending on request parameter: `stop and start` -> number of addresses generated -> number of balances returned, because originally it was testing several combinations of start and stop.
Changed to more static version by seeding response [11,11]